needle.pm

```diff
@@ -121,21 +121,13 @@ sub new {
     }
 
     # one match is mandatory
-    unless ($gotmatch) {
-        warn "$jsonfile missing match area\n";
-        return;
-    }
+    warn "$jsonfile missing match area\n" && return unless $gotmatch;
 
     $self->{name} = basename($jsonfile, '.json');
     my $png = $self->{png} || $self->{name} . ".png";
 
     $self->{png} = File::Spec->catdir(dirname($jsonfile), $png);
-
-    if (!-s $self->{png}) {
-        warn "Can't find $self->{png}";
-        return;
-    }
-
+    warn "Can't find $self->{png}" && return unless -s $self->{png};
```

Comment:
Where does `warn` come from? I just wanted to cross-check whether it actually returns a truthy value all the time. Otherwise I'd refrain from depending on the return value of the function.

Reply:
It's Perl's core "warn" and it always returns a truthy value